apps/gitness/src/pages-v2/pull-request/hooks/usePRChecksDecision.ts

```diff
@@ -0,0 +1,181 @@
+import { useEffect, useMemo, useState } from 'react'
+
+import pluralize from 'pluralize'
+
+import { TypesPullReq, TypesRepository, useChecksPullReqQuery } from '@harnessio/code-service-client'
+import { ExecutionState } from '@harnessio/views'
+
+import { determineStatusMessage, generateStatusSummary } from '../../../pages/pull-request/utils'
+
+interface Check {
+  check: {
+    status: ExecutionState
+  }
+}
+
+interface Data {
+  checks: Check[]
+}
+export function usePRChecksDecision({
+  repoMetadata,
+  pullReqMetadata
+}: {
+  repoMetadata: TypesRepository | undefined
+  pullReqMetadata: TypesPullReq | undefined
```

Comment:
Can we make these optional instead?
```suggestion
  repoMetadata?: TypesRepository
  pullReqMetadata?: TypesPullReq
```